tests/sql/operators/test_cleanup.py

```diff
@@ -0,0 +1,208 @@
+import pathlib
+
+import pandas
+import pytest
+from airflow.models.taskinstance import TaskInstance
+from airflow.operators.bash import BashOperator
+from airflow.utils.state import State
+
+import astro.sql as aql
+from astro.constants import Database
+from astro.files import File
+from astro.sql.operators.cleanup import CleanupOperator
+from astro.sql.table import Table
+from tests.sql.operators import utils as test_utils
+
+CWD = pathlib.Path(__file__).parent
+
+DEFAULT_FILEPATH = str(pathlib.Path(CWD.parent.parent, "data/sample.csv").absolute())
+SQLITE_ONLY = [
+    {
+        "database": Database.SQLITE,
+    },
+]
+SUPPORTED_DATABASES = [
+    {
+        "database": Database.SQLITE,
+    },
+    {
+        "database": Database.POSTGRES,
+    },
+    {
+        "database": Database.BIGQUERY,
+    },
+    {
+        "database": Database.SNOWFLAKE,
+    },
+]
+SUPPORTED_DATABASES_WITH_FILE = [
+    dict(x, **{"file": File(DEFAULT_FILEPATH)}) for x in SUPPORTED_DATABASES
+]
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    SUPPORTED_DATABASES_WITH_FILE,
+    indirect=True,
+    ids=["sqlite", "postgres", "bigquery", "snowflake"],
+)
+def test_cleanup_one_table(database_table_fixture):
+    db, test_table = database_table_fixture
+    assert db.table_exists(test_table)
+    a = aql.cleanup([test_table])
+    a.execute({})
+    assert not db.table_exists(test_table)
+
+
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    SUPPORTED_DATABASES,
+    indirect=True,
+    ids=["sqlite", "postgres", "bigquery", "snowflake"],
+)
+@pytest.mark.parametrize(
+    "multiple_tables_fixture",
+    [
+        {
+            "items": [
+                {
+                    "table": Table(name="non_temp_table"),
+                    "file": File(DEFAULT_FILEPATH),
+                },
+                {
+                    "table": Table(),
+                    "file": File(DEFAULT_FILEPATH),
+                },
+            ]
+        }
+    ],
+    indirect=True,
+    ids=["named_table"],
+)
+def test_cleanup_non_temp_table(database_table_fixture, multiple_tables_fixture):
+    db, _ = database_table_fixture
+    test_table, test_temp_table = multiple_tables_fixture
+    assert db.table_exists(test_table)
+    assert db.table_exists(test_temp_table)
+    test_table.conn_id = db.conn_id
+    test_temp_table.conn_id = db.conn_id
+    a = aql.cleanup([test_table, test_temp_table])
+    a.execute({})
+    assert db.table_exists(test_table)
+    assert not db.table_exists(test_temp_table)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    SUPPORTED_DATABASES_WITH_FILE,
+    indirect=True,
+    ids=["sqlite", "postgres", "bigquery", "snowflake"],
+)
+def test_cleanup_non_table(database_table_fixture):
```

Comment:
@kaxil that's fair. I've change the test to only run against sqlite (As that's essentially the same as mocking since it's all local/runs super quickly).